### PR TITLE
Changes ASSERT_TRUE to ASSERT_EQ for lots of cases.

### PR DESCRIPTION
**Pull request by tobsan**
Also removes all asserts in the code and replaces them with
proper error handling.

Signed-off-by: Tobias Olausson tobias.olausson@pelagicore.com

### CODE REVIEW

File: common/softwarecontainer-common.cpp [hunk 1]
```diff
@@ -35,6 +35,8 @@ LOG_DECLARE_DEFAULT_CONTEXT(defaultLogContext, "MAIN", "Main context");
 struct stat getStat(const std::string &path)
 {
     struct stat st;
+    memset(&st, 0, sizeof(st));
+
```
Comment (DunderRoffe) [inline common/softwarecontainer-common.cpp]: Should be in a separate commit

**Review by DunderRoffe — CHANGES_REQUESTED** [review]

**Review by thenor — APPROVED** [review]
Looks good to me.

**Review by DunderRoffe — APPROVED** [review]
